src/nanoarrow/nanoarrow.h

```diff
@@ -454,6 +454,73 @@ ArrowErrorCode ArrowSchemaViewInit(struct ArrowSchemaView* schema_view,
 
 /// }@
 
+/// \defgroup nanoarrow-buffer-builder Growable buffer builders
+
+/// \brief An owning mutable view of a buffer
+struct ArrowBuffer {
+  /// \brief A pointer to the start of the buffer
+  ///
+  /// If capacity_bytes is 0, this value may be NULL.
+  uint8_t* data;
+
+  /// \brief The size of the buffer in bytes
+  int64_t size_bytes;
+
+  /// \brief The capacity of the buffer in bytes
+  int64_t capacity_bytes;
```

Comment:
Slightly disruptive proposal: how about adding an `item_width` and change all sizes and capacities to be expressed in number of items? Or would we want to do that as a separate primitive (or not at all)?

Reply:
Yes, that would be the idea. You'd use that just like you use a TypedBufferBuilder, but without type templating which doesn't really exist in C :-)

Reply:
Yes, that's also a possibility. Let's move on then :-)